core/src/main/scala/kafka/admin/AdminUtils.scala

```diff
@@ -661,9 +667,83 @@ object AdminUtils extends Logging with AdminUtilities {
       .map(entityPath => (entityPath, fetchEntityConfig(zkUtils, rootEntityType, entityPath))).toMap
   }
 
+  def fetchTopicMetadata(topics: Set[String], brokers: Seq[BrokerEndPoint], properties: Properties)
```

Comment:
On second thought, this should probably not be here. `AdminUtils` is the place for methods that interact with `ZK`. We could add it to the `ClientUtils` in the client module maybe (that makes it clear that we are communicating with Kafka via the protocol).